--- gdtoa-1.0/gdtoa.h	2012-09-19 10:39:14.178175086 +0200
+++ gdtoa-1.0/gdtoa.h	2012-09-19 10:51:30.042159541 +0200
@@ -32,6 +32,12 @@
 #ifndef GDTOA_H_INCLUDED
 #define GDTOA_H_INCLUDED
 
+#include <stdlib.h>
+#ifndef __THROW
+#define __THROW
+#define __UNDEF_THROW
+#endif
+
 #include "arith.h"
 
 #ifndef Long
@@ -107,8 +113,8 @@
 extern char* gdtoa ANSI((FPI *fpi, int be, ULong *bits, int *kindp,
 			int mode, int ndigits, int *decpt, char **rve));
 extern void freedtoa ANSI((char*));
-extern float  strtof ANSI((CONST char *, char **));
-extern double strtod ANSI((CONST char *, char **));
+extern float  strtof ANSI((CONST char *, char **)) __THROW;
+extern double strtod ANSI((CONST char *, char **)) __THROW;
 extern int strtodg ANSI((CONST char*, char**, FPI*, Long*, ULong*));
 
 extern char*	g_ddfmt  ANSI((char*, double*, int, unsigned));
@@ -150,4 +156,11 @@
 #ifdef __cplusplus
 }
 #endif
+
+
+#ifdef __UNDEF_THROW
+#undef __THROW
+#undef __UNDEF_THROW
+#endif
+
 #endif /* GDTOA_H_INCLUDED */
