--- gdtoa-1.0/gdtoa.h	2012-09-19 10:39:14.178175086 +0200
+++ gdtoa-1.0/gdtoa.h	2012-09-19 10:51:30.042159541 +0200
@@ -32,6 +32,12 @@
 #ifndef GDTOA_H_INCLUDED
 #define GDTOA_H_INCLUDED
 
+#include <stdlib.h>
+#ifndef __THROW
+#define __THROW
+#define __UNDEF_THROW
+#endif
+
 #include "arith.h"
 
 #ifndef Long
@@ -107,8 +113,8 @@
 extern char* gdtoa ANSI((FPI *fpi, int be, ULong *bits, int *kindp,
 			int mode, int ndigits, int *decpt, char **rve));
 extern void freedtoa ANSI((char*));
-extern float  strtof ANSI((CONST char *, char **));
-extern double strtod ANSI((CONST char *, char **));
+extern float  strtof ANSI((CONST char *, char **)) __THROW;
+extern double strtod ANSI((CONST char *, char **)) __THROW;
 extern int strtodg ANSI((CONST char*, char**, FPI*, Long*, ULong*));
 
 extern char*	g_ddfmt  ANSI((char*, double*, int, unsigned));
@@ -150,4 +156,11 @@
 #ifdef __cplusplus
 }
 #endif
+
+
+#ifdef __UNDEF_THROW
+#undef __THROW
+#undef __UNDEF_THROW
+#endif
+
 #endif /* GDTOA_H_INCLUDED */
--- gdtoa-1.0/gdtoaimp.h	2012-09-19 15:57:30.785771662 +0200
+++ gdtoa-1.0/gdtoaimp.h	2012-09-19 15:58:04.641770943 +0200
@@ -508,6 +508,11 @@
 #define trailz trailz_D2A
 #define ulp ulp_D2A
 
+#ifndef __THROW
+#define __THROW
+#define __UNDEF_THROW
+#endif
+
  extern char *dtoa_result;
  extern CONST double bigtens[], tens[], tinytens[];
  extern unsigned char hexdig[];
@@ -551,11 +556,16 @@
  extern Bigint *set_ones ANSI((Bigint*, int));
  extern char *strcp ANSI((char*, const char*));
  extern int strtoIg ANSI((CONST char*, char**, FPI*, Long*, Bigint**, int*));
- extern double strtod ANSI((const char *s00, char **se));
+ extern double strtod ANSI((const char *s00, char **se)) __THROW;
  extern Bigint *sum ANSI((Bigint*, Bigint*));
  extern int trailz ANSI((Bigint*));
  extern double ulp ANSI((double));
 
+#ifdef __UNDEF_THROW
+#undef __THROW
+#undef __UNDEF_THROW
+#endif
+
 #ifdef __cplusplus
 }
 #endif
