--- soplex-4.0.2/make/make.install	2019-06-28 15:48:00.000000000 +0200
+++ soplex-4.0.2/make/make.install	2020-03-13 16:12:41.131803815 +0100
@@ -29,7 +29,9 @@
 ifneq ($(INSTALLDIR),)
 		@echo "-> install SoPlex headers into INSTALLDIR <$(INSTALLDIR)>"
 		@-install -m 755 -d $(INSTALLDIR)/$(INCLUDEDIR)/
-		@-install -m 644 -p $(LIBSRCHEADER) $(INSTALLDIR)/$(INCLUDEDIR)/
+		@-install -m 755 -d $(INSTALLDIR)/$(INCLUDEDIR)/soplex/
+		@-install -m 644 -p $(SRCDIR)/soplex.h $(INSTALLDIR)/$(INCLUDEDIR)/
+		@-install -m 644 -p $(filter-out $(SRCDIR)/soplex.h,$(LIBSRCHEADER)) $(INSTALLDIR)/$(INCLUDEDIR)/soplex/
 else
 		@echo "no installation directory specified - use make install INSTALLDIR=<dir>, e.g., make install INSTALLDIR=/usr/local"
 endif
--- soplex-4.0.2/src/soplex/spxdefines.h	2019-06-28 15:48:01.000000000 +0200
+++ soplex-4.0.2/src/soplex/spxdefines.h	2025-02-08 15:05:10.093119564 +0100
@@ -22,7 +22,7 @@
  * \ref soplex::SPxOut::INFO3 "INFO3" are set.
  * If \c NDEBUG is not defined, the code within \#TRACE is used.
  * If \c SOPLEX_DEBUG is defined, the code within
- * \ref soplex::SPxOut::DEBUG "DEBUG" is also used.
+ * \ref soplex::SPxOut::_SOLEX_DEBUG "DEBUG" is also used.
  *
  * If \c WITH_LONG_DOUBLE is defined, all Real numbers are of type
  * long double instead of just double.
--- soplex-4.0.2/src/soplex/spxout.cpp	2019-06-28 15:48:01.000000000 +0200
+++ soplex-4.0.2/src/soplex/spxout.cpp	2025-02-08 15:04:23.499336072 +0100
@@ -28,7 +28,7 @@
    m_streams = new(m_streams) std::ostream*[INFO3 + 1];
    m_streams[ ERROR ] = m_streams[ WARNING ] = &std::cerr;
 
-   for(int i = DEBUG; i <= INFO3; ++i)
+   for(int i = _SOLEX_DEBUG; i <= INFO3; ++i)
       m_streams[ i ] = &std::cout;
 }
 
@@ -44,7 +44,7 @@
 {
    m_verbosity = base.m_verbosity;
 
-   for(int i = DEBUG; i <= INFO3; ++i)
+   for(int i = _SOLEX_DEBUG; i <= INFO3; ++i)
       m_streams[ i ] = base.m_streams[ i ];
 
    return *this;
@@ -58,7 +58,7 @@
    m_streams = new(m_streams) std::ostream*[INFO3 + 1];
    m_streams[ ERROR ] = m_streams[ WARNING ] = rhs.m_streams[ERROR];
 
-   for(int i = DEBUG; i <= INFO3; ++i)
+   for(int i = _SOLEX_DEBUG; i <= INFO3; ++i)
       m_streams[ i ] = rhs.m_streams[ i ];
 }
 
--- soplex-4.0.2/src/soplex/spxout.h	2019-06-28 15:48:01.000000000 +0200
+++ soplex-4.0.2/src/soplex/spxout.h	2025-02-08 15:04:41.722933555 +0100
@@ -76,7 +76,7 @@
       // If you change this, change the implementation as well.
       ERROR    = 0,
       WARNING  = 1,
-      DEBUG    = 2,
+      _SOLEX_DEBUG    = 2,
       INFO1    = 3,
       INFO2    = 4,
       INFO3    = 5
--- soplex-4.0.2/src/soplex.cpp	2019-06-28 15:48:00.000000000 +0200
+++ soplex-4.0.2/src/soplex.cpp	2025-02-08 15:03:35.577552988 +0100
@@ -5974,7 +5974,7 @@
          break;
 
       case 2:
-         spxout.setVerbosity(SPxOut::DEBUG);
+         spxout.setVerbosity(SPxOut::_SOLEX_DEBUG);
          break;
 
       case 3:
