--- gaol-4.2.0/gaol/gaol_allocator.h	2012-03-06 18:08:58.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_allocator.h	2019-06-21 11:08:57.446030219 +0200
@@ -89,7 +89,7 @@
 			return std::numeric_limits<std::size_t>::max() / sizeof ( T );
 		}
 
-		pointer allocate ( size_type num, const void* = 0 ) throw ( std::bad_alloc )
+		pointer allocate ( size_type num, const void* = 0 )
 		{
 			void *buf;
 			if ( MEMALIGN( buf,16,num*sizeof ( value_type ) ) )   // Allocation error?
--- gaol-4.2.0/gaol/gaol_double_op_apmathlib.h	2012-03-06 18:08:58.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_double_op_apmathlib.h	2019-06-21 12:25:08.036229637 +0200
@@ -112,7 +112,7 @@
     	for (;;) {
       	if (odd(n)) {
 				n >>= 1;
-				y = opposite(opposite(y)*z);
+				y = gaol_opposite(gaol_opposite(y)*z);
 				if (n == 0) {
 	  				GAOL_RND_LEAVE();
 	  				return y;
@@ -120,7 +120,7 @@
       	} else {
 				n>>=1;
       	}
-      	z = opposite(opposite(z)*z);
+      	z = gaol_opposite(gaol_opposite(z)*z);
     	}
 	}
 
@@ -134,9 +134,9 @@
 	return ipow_dn(d,e);
       } else { // d < 0
 	if (even(e)) {
-	  return ipow_dn(opposite(d),e);
+	  return ipow_dn(gaol_opposite(d),e);
 	} else { // odd(e)
-	  return opposite(ipow_up(opposite(d),e));
+	  return gaol_opposite(ipow_up(gaol_opposite(d),e));
 	}
       }
     }
@@ -151,9 +151,9 @@
 	return ipow_up(d,e);
       } else { // d < 0
 	if (even(e)) {
-	  return ipow_up(opposite(d),e);
+	  return ipow_up(gaol_opposite(d),e);
 	} else { // odd(e)
-	  return opposite(ipow_dn(opposite(d),e));
+	  return gaol_opposite(ipow_dn(gaol_opposite(d),e));
 	}
       }
     }
--- gaol-4.2.0/gaol/gaol_double_op_crlibm.h	2012-03-06 18:08:58.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_double_op_crlibm.h	2019-06-21 12:25:28.901056220 +0200
@@ -88,7 +88,7 @@
     	for (;;) {
       	if (odd(n)) {
 			n >>= 1;
-			y = opposite(opposite(y)*z);
+			y = gaol_opposite(gaol_opposite(y)*z);
 			if (n == 0) {
 	  			GAOL_RND_LEAVE();
 	  			return y;
@@ -96,7 +96,7 @@
       	} else {
 			n>>=1;
       	}
-      	z = opposite(opposite(z)*z);
+      	z = gaol_opposite(gaol_opposite(z)*z);
     	}
 	}
 
@@ -110,9 +110,9 @@
 	return ipow_dn(d,e);
       } else { // d < 0
 	if (even(e)) {
-	  return ipow_dn(opposite(d),e);
+	  return ipow_dn(gaol_opposite(d),e);
 	} else { // odd(e)
-	  return opposite(ipow_up(opposite(d),e));
+	  return gaol_opposite(ipow_up(gaol_opposite(d),e));
 	}
       }
     }
@@ -127,9 +127,9 @@
 	return ipow_up(d,e);
       } else { // d < 0
 	if (even(e)) {
-	  return ipow_up(opposite(d),e);
+	  return ipow_up(gaol_opposite(d),e);
 	} else { // odd(e)
-	  return opposite(ipow_dn(opposite(d),e));
+	  return gaol_opposite(ipow_dn(gaol_opposite(d),e));
 	}
       }
     }
--- gaol-4.2.0/gaol/gaol_double_op_m.h	2009-03-01 14:53:01.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_double_op_m.h	2019-06-21 12:25:46.817473387 +0200
@@ -115,7 +115,7 @@
     	for (;;) {
       	if (odd(n)) {
 				n >>= 1;
-				y = opposite(opposite(y)*z);
+				y = gaol_opposite(gaol_opposite(y)*z);
 				if (n == 0) {
 	  				GAOL_RND_LEAVE();
 	  				return y;
@@ -123,7 +123,7 @@
       	} else {
 				n>>=1;
       	}
-      	z = opposite(opposite(z)*z);
+      	z = gaol_opposite(gaol_opposite(z)*z);
     	}
 	}
 
@@ -137,9 +137,9 @@
 	return ipow_dn(d,e);
       } else { // d < 0
 	if (even(e)) {
-	  return ipow_dn(opposite(d),e);
+	  return ipow_dn(gaol_opposite(d),e);
 	} else { // odd(e)
-	  return opposite(ipow_up(opposite(d),e));
+	  return gaol_opposite(ipow_up(gaol_opposite(d),e));
 	}
       }
     }
@@ -154,9 +154,9 @@
 	return ipow_up(d,e);
       } else { // d < 0
 	if (even(e)) {
-	  return ipow_up(opposite(d),e);
+	  return ipow_up(gaol_opposite(d),e);
 	} else { // odd(e)
-	  return opposite(ipow_dn(opposite(d),e));
+	  return gaol_opposite(ipow_dn(gaol_opposite(d),e));
 	}
       }
     }
--- gaol-4.2.0/gaol/gaol_interval2f.cpp	2012-03-06 18:08:58.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_interval2f.cpp	2019-06-21 11:10:43.842842834 +0200
@@ -40,7 +40,7 @@
 	}
 
 
-	void* interval2f::operator new(size_t sz) throw(std::bad_alloc)
+	void* interval2f::operator new(size_t sz)
 	{
 		void *buf;
 		if (MEMALIGN(buf,16,sz)) { // error ?
@@ -54,7 +54,7 @@
         free(p);
     }
 
-	void* interval2f::operator new[](size_t sz) throw(std::bad_alloc)
+	void* interval2f::operator new[](size_t sz)
 	{
 		void *buf;
 		if (MEMALIGN(buf,16,sz)) { // error ?
@@ -70,7 +70,7 @@
         free(p);
     }
 
-	void* interval2f::operator new(size_t sz, void *p) throw(std::bad_alloc)
+	void* interval2f::operator new(size_t sz, void *p)
 	{
 		if ((unsigned long long)p % 16 != 0) {
 			throw std::bad_alloc();
--- gaol-4.2.0/gaol/gaol_interval2f.h	2012-03-06 18:08:58.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_interval2f.h	2019-06-21 11:10:17.773582939 +0200
@@ -94,13 +94,13 @@
 
 
       // Allocators defined to enforce alignment on a 16 bytes boundary
-      __GAOL_PUBLIC__ void *operator new(size_t sz) throw(std::bad_alloc);
+      __GAOL_PUBLIC__ void *operator new(size_t sz);
       __GAOL_PUBLIC__ void operator delete(void *p) throw();
 
-      __GAOL_PUBLIC__ void *operator new[](size_t sz) throw(std::bad_alloc);
+      __GAOL_PUBLIC__ void *operator new[](size_t sz);
       __GAOL_PUBLIC__ void operator delete[](void *p) throw();
 
-      __GAOL_PUBLIC__ void *operator new(size_t sz, void *p) throw(std::bad_alloc);
+      __GAOL_PUBLIC__ void *operator new(size_t sz, void *p);
       __GAOL_PUBLIC__ void operator delete(void *p, void *place) throw();
 
 			friend interval2f pow(const interval2f& I, int n);
--- gaol-4.2.0/gaol/gaol_interval.h	2015-05-21 09:00:45.000000000 +0200
+++ gaol-4.2.0/gaol/gaol_interval.h	2019-06-21 11:08:27.704865552 +0200
@@ -32,6 +32,7 @@
 
 #include <cmath>
 #include <iosfwd>
+#include <ios> /* for std::streamsize, patch from ibex */
 #include <string>
 #include <limits>
 #include "gaol/gaol_config.h"
@@ -262,13 +263,13 @@
     static const __m128d m128d_20mask;
 
     // Allocators defined to enforce alignment on a 16 bytes boundary
-    void *operator new(size_t sz) throw(std::bad_alloc);
+    void *operator new(size_t sz);
     void operator delete(void *p) throw();
 
-    void *operator new[](size_t sz) throw(std::bad_alloc);
+    void *operator new[](size_t sz);
     void operator delete[](void *p) throw();
 
-    void *operator new(size_t sz, void *p) throw(std::bad_alloc);
+    void *operator new(size_t sz, void *p);
     void operator delete(void *p, void *place) throw();
 
 #else // !USING_SSE2_INSTRUCTIONS
--- gaol-4.2.0/gaol/gaol_interval_sse.cpp	2012-03-06 18:08:58.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_interval_sse.cpp	2019-06-21 11:57:38.571567641 +0200
@@ -35,7 +35,7 @@
 	const __m128d interval::m128_nan = _mm_set1_pd(std::numeric_limits<double>::quiet_NaN());
 	const __m128d interval::m128_infinf = _mm_set1_pd(std::numeric_limits<double>::infinity());
 
-	void* interval::operator new(size_t sz) throw(std::bad_alloc)
+	void* interval::operator new(size_t sz)
 	{
 		void *buf;
 		if (MEMALIGN(buf,16,sz)) { // error ?
@@ -49,7 +49,7 @@
         free(p);
     }
 
-	void* interval::operator new[](size_t sz) throw(std::bad_alloc)
+	void* interval::operator new[](size_t sz)
 	{
 		void *buf;
 		if (MEMALIGN(buf,16,sz)) { // error ?
@@ -65,8 +65,7 @@
         free(p);
     }
 
-	void* interval::operator new(size_t sz, void *p) throw(std::bad_alloc)
-	{
+	void* interval::operator new(size_t sz, void *p)	{
 		if ((unsigned long long)p % 16 != 0) {
 			throw std::bad_alloc();
 		}
--- gaol-4.2.0/gaol/gaol_port.h	2012-03-06 18:08:58.000000000 +0100
+++ gaol-4.2.0/gaol/gaol_port.h	2019-06-21 12:24:29.653533877 +0200
@@ -89,9 +89,9 @@
 
 #if HAVE_ROUNDING_MATH_OPTION
   INLINE double f_negate_simple(double x) { return -x; }
-#  define opposite(x) f_negate_simple(x)
+#  define gaol_opposite(x) f_negate_simple(x)
 #else
-#  define opposite(x) f_negate(x)
+#  define gaol_opposite(x) f_negate(x)
 #endif // HAVE_ROUNDING_MATH_OPTION
 
   /*!
@@ -145,8 +145,7 @@
 #endif
 
 #ifndef GAOL_NAN
-  static uintdouble NaN_val = {{IFBIGENDIAN(0x7ff80000, 0x0)}};
-#define GAOL_NAN (gaol::NaN_val.d)
+#define GAOL_NAN (((uintdouble) {{IFBIGENDIAN(0x7ff80000, 0x0)}}).d)
 #endif
 
   // #define GAOL_INFINITY std::numeric_limits<double>::infinity()
--- gaol-4.2.0/gdtoa/dtoa.c	2009-02-17 17:39:38.000000000 +0100
+++ gaol-4.2.0/gdtoa/dtoa.c	2019-06-21 10:39:11.771631058 +0200
@@ -73,7 +73,7 @@
 #define Rounding Flt_Rounds
 #endif
 
-int Flt_Rounds;
+int Flt_Rounds = 1; /* round to the nearest by default, patched by ibex */
 
  char *
 dtoa
