--- gaol-4.2.0/gaol/gaol_port.h	2016-07-28 13:51:17.602655218 +0200
+++ gaol-4.2.0/gaol/gaol_port.h	2016-07-29 10:22:36.200271112 +0200
@@ -37,9 +37,7 @@
 #include "gaol/gaol_config.h"
 #include "gaol/gaol_limits.h"
 
-#if HAVE_FINITE
-#  include <cmath>
-#endif
+#include <cmath>
 
 // Alignment on an 'nbytes' bytes boundary
 #if defined(_MSC_VER)
@@ -193,11 +191,7 @@
    */
   INLINE int is_finite(double d)
   {
-#if HAVE_FINITE
-    return finite(d);
-#else
     return (d==d && d!=GAOL_INFINITY && d!=-GAOL_INFINITY);
-#endif
   }
 
 
